zerver/actions/message_edit.py

```diff
@@ -160,7 +161,15 @@ def maybe_send_resolve_topic_notifications(
         RESOLVED_TOPIC_PREFIX
     ) and not new_topic_name.startswith(RESOLVED_TOPIC_PREFIX)
 
-    if not topic_resolved and not topic_unresolved:
+    if (
+        (not topic_resolved and not topic_unresolved)
+        or (
+            topic_resolved
+            and new_topic_name[2:] != old_topic_name
+            and len(new_topic_name) != MAX_TOPIC_NAME_LENGTH
+        )
```

Comment:
This doesn't work in the case if
* old topic name = "topic one"
* New topic name = "✔ topic one"

and we move a message from old to new topic. Weird edge case but possible.

In any case, we shouldn't use a hardcoded integer **2**. Instead, `a_meaningful_variable_name = len(RESOLVED_TOPIC_PREFIX)` would be better.

Reply:
I think, there can me more edge cases here depending on what topic name someone uses !

Reply:
Something around `propagate_mode = change_all` should be a better approach.

Reply:
Hey, thanks for your review @prakhar1144. The case that you mentioned is a very particular case, where for some reason there is two versions of the same topic, one resolved and another unresolved. If we move a message from the unresolved version to the resolved version or vice-versa, the program will normally detect that the topic was resolved/unresolved because we check this by comparing their names. I am not sure that this can fixed with `propagate_mode = change_all`, I think we would need to change the entire logic behind resolved and unresolved topics. I am also pretty confident that there is no other edge cases as the changes I have made are pretty simple and I have thought about what could go wrong extensively. Please note that the changes consist on simply not sending the notification if the message was moved to a different topic, however in your very particular case the new topic is considered the same because it has the same name.

About the hardcoded integer 2, I completely agree! I can commit a new version with a variable replacing the 2 as you asked for, and maybe with also a better comment inside the if case, explaining it better. Please let me know if this is ok, or if you will only accept a PR that is able to fix the particular case that you mentioned.

Reply:
Let'd discuss in [chat.zulip.org](https://chat.zulip.org) Input from everyone would be helpful.